tests/entities/test_experiment.py

```diff
@@ -1,50 +1,67 @@
 from tests.helper_functions import random_int, random_file
 
 
 class TestExperiment(unittest.TestCase):
-    def _check(self, exp, exp_id, name, location, lifecyle_stage):
+    def _check(self, exp, exp_id, name, location, lifecyle_stage, creation_time, last_update_time):
         self.assertIsInstance(exp, Experiment)
         self.assertEqual(exp.experiment_id, exp_id)
         self.assertEqual(exp.name, name)
         self.assertEqual(exp.artifact_location, location)
         self.assertEqual(exp.lifecycle_stage, lifecyle_stage)
+        self.assertEqual(exp.creation_time, creation_time)
+        self.assertEqual(exp.last_update_time, last_update_time)
 
     def test_creation_and_hydration(self):
         exp_id = str(random_int())
         name = "exp_%d_%d" % (random_int(), random_int())
         lifecycle_stage = LifecycleStage.ACTIVE
         location = random_file(".json")
+        creation_time = int(time.time())
+        last_update_time = int(time.time())
 
-        exp = Experiment(exp_id, name, location, lifecycle_stage)
-        self._check(exp, exp_id, name, location, lifecycle_stage)
+        exp = Experiment(
+            exp_id,
+            name,
+            location,
+            lifecycle_stage,
+            creation_time=creation_time,
+            last_update_time=last_update_time,
+        )
+        self._check(exp, exp_id, name, location, lifecycle_stage, creation_time, last_update_time)
 
         as_dict = {
             "experiment_id": exp_id,
             "name": name,
             "artifact_location": location,
             "lifecycle_stage": lifecycle_stage,
             "tags": {},
+            "creation_time": creation_time,
+            "last_update_time": last_update_time,
         }
         self.assertEqual(dict(exp), as_dict)
 
         proto = exp.to_proto()
         exp2 = Experiment.from_proto(proto)
-        self._check(exp2, exp_id, name, location, lifecycle_stage)
+        self._check(exp2, exp_id, name, location, lifecycle_stage, creation_time, last_update_time)
 
         exp3 = Experiment.from_dictionary(as_dict)
-        self._check(exp3, exp_id, name, location, lifecycle_stage)
+        self._check(exp3, exp_id, name, location, lifecycle_stage, creation_time, last_update_time)
 
     def test_string_repr(self):
         exp = Experiment(
             experiment_id=0,
             name="myname",
             artifact_location="hi",
             lifecycle_stage=LifecycleStage.ACTIVE,
+            creation_time=1661624204,
+            last_update_time=1661624204,
         )
         assert (
-            str(exp) == "<Experiment: artifact_location='hi', experiment_id=0, "
-            "lifecycle_stage='active', name='myname', tags={}>"
+            str(exp)
+            == "<Experiment: artifact_location='hi', creation_time=1661624204, experiment_id=0, "
+            "last_update_time=1661624204, lifecycle_stage='active', name='myname', tags={}>"
```

Comment:
Do we have tests for `restore_experiment` and `delete_experiment`?

Reply:
There are no explicit checks for last_update_time in those scenarios. I am working on it.

Reply:
@harupy I am envisioning something like below. do think just checking `last_update_time != None` is good enough.

```
        exp1 = fs.get_experiment(exp_id)

        // ms might be too low, now() - 5 second ???
        exp1._set_last_update_time(int(time.time() * 1000))

        fs.restore_experiment(exp_id)
        self.assertEqual(fs.get_experiment(exp_id).name, new_name)
        assert restored_1.last_update_time > exp1.last_update_time
```

Reply:
Can we do this?

```python
        exp_id = fs.create_experiment("name")
        exp = fs.get_experiment(exp_id)
        time.sleep(0.01)  # 10 ms
        fs.restore_experiment(exp_id)
        assert fs.get_experiment(exp_id).last_update_time > exp.last_update_time
```

Reply:
I was thinking sleep might slow down tests. But given the number of test cases for experiments, I think it should be fine.

Reply:
A short sleep (< 10 ms) is fine, but a long sleep is not.